universe/src/storage/cassandra.rs

```diff
@@ -0,0 +1,420 @@
+                                .unwrap()
+                                .as_str_name()
+                                .to_string(),
+                        ),
+                        Some(proto::universe::region::Cloud::OtherCloud(other)) => Some(other),
+                        None => None,
+                    },
+                }),
+            },
+            base_key_ranges: base_key_range_requests
+                .into_iter()
+                .map(|range| SerializedKeyRange {
+                    base_range_uuid: Uuid::from_str(&range.base_range_uuid).unwrap(),
+                    lower_bound_inclusive: Some(range.lower_bound_inclusive),
+                    upper_bound_exclusive: Some(range.upper_bound_exclusive),
+                })
+                .collect(),
+        }
+    }
+
+    /// Convert into a KeyspaceInfo proto.
+    fn to_keyspace_info(self) -> KeyspaceInfo {
+        let primary_zone = Zone {
+            name: self.primary_zone.name,
+            region: self
+                .primary_zone
+                .region
+                .map(|region| proto::universe::Region {
+                    name: region.name,
+                    cloud: region.cloud.map(|cloud| match cloud.as_str() {
+                        "AWS" => proto::universe::region::Cloud::PublicCloud(
+                            proto::universe::Cloud::Aws as i32,
+                        ),
+                        "AZURE" => proto::universe::region::Cloud::PublicCloud(
+                            proto::universe::Cloud::Azure as i32,
+                        ),
+                        "GCP" => proto::universe::region::Cloud::PublicCloud(
+                            proto::universe::Cloud::Gcp as i32,
+                        ),
+                        other => proto::universe::region::Cloud::OtherCloud(other.to_string()),
+                    }),
+                }),
+        };
+        let base_key_ranges = self
+            .base_key_ranges
+            .into_iter()
+            .map(|range| KeyRange {
+                base_range_uuid: range.base_range_uuid.to_string(),
+                lower_bound_inclusive: range.lower_bound_inclusive.unwrap_or_default(),
+                upper_bound_exclusive: range.upper_bound_exclusive.unwrap_or_default(),
+            })
+            .collect();
+        KeyspaceInfo {
+            keyspace_id: self.keyspace_id.to_string(),
+            name: self.name,
+            namespace: self.namespace,
+            primary_zone: Some(primary_zone),
+            base_key_ranges,
+        }
+    }
+}
+
+impl Cassandra {
+    pub async fn new(known_node: String) -> Self {
+        let session = SessionBuilder::new()
+            .known_node(known_node)
+            .build()
+            .await
+            .unwrap();
+        Self { session }
+    }
+}
+
+impl Storage for Cassandra {
+    async fn create_keyspace(
+        &self,
+        name: &str,
+        namespace: &str,
+        primary_zone: Option<Zone>,
```

Comment:
why take an option here then unwrap it? I think let's require this to be a Zone and unwrap at the caller.

Reply:
oops, that's a leftover